### ISSUE TITLE
declare/constant interferes with assume

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2997]
Imported from SourceForge on 2024-07-07 08:35:46
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2013-02-12 02:11:11
Original: https://sourceforge.net/p/maxima/bugs/2547

---

(%i14) assume(q>r);
(%o14)          [q > r]
(%i15) is(q>r);
(%o15)            true
(%i16) declare(q,constant);
(%o16)            done
(%i17) is(q>r);
(%o17)            true
(%i18) declare(r,constant);
(%o18)            done
(%i19) is(q>r);
(%o19)          unknown            <<<<!!

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 08:35:47
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2013-02-12 02:12:07
Original: https://sourceforge.net/p/maxima/bugs/2547/#d118

---

Maxima version: 5.25.1
Maxima build date: 10:2 9/6/2011
Host type: i686-pc-mingw32
Lisp implementation type: GNU Common Lisp (GCL)
Lisp implementation version: GCL 2.6.8

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 08:35:51
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2013-02-12 02:16:50
Original: https://sourceforge.net/p/maxima/bugs/2547/#176a

---

Hmm, why was I running 5.25?  Anyway, same problem in:

Maxima version: "5.28.0-2"
Maxima build date: "2012-08-27 23:16:48"
Host type: "i686-pc-mingw32"
Lisp implementation type: "GNU Common Lisp (GCL)"
Lisp implementation version: "GCL 2.6.8"

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 08:35:54
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2013-02-16 11:37:30
Original: https://sourceforge.net/p/maxima/bugs/2547/#cfbc

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 08:35:58
Created by **[rswarbrick](https://sourceforge.net/u/rswarbrick/)** on 2013-02-16 11:37:30
Original: https://sourceforge.net/p/maxima/bugs/2547/#92ca

---

Hopefully fixed by commit [acedfa0][1]

~~~~
Maxima 5.28.0_188_gd20c5d6 http://maxima.sourceforge.net
using Lisp SBCL 1.0.57.0.debian
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) assume (q>r);
(%o1)                               [q > r]
(%i2) is (q > r);
(%o2)                                true
(%i3) declare (q, constant);
(%o3)                                done
(%i4) is (q > r);
(%o4)                                true
(%i5) declare (r, constant);
(%o5)                                done
(%i6) is (q > r);
(%o6)                                true
~~~~

[1]: https://sourceforge.net/p/maxima/code/ci/acedfa0cf33cd1b26d3e65c0f259c0fa83fb997f/
